src/ray/object_manager/object_directory.h

```diff
@@ -0,0 +1,134 @@
+#ifndef RAY_OBJECT_MANAGER_OBJECT_DIRECTORY_H
+#define RAY_OBJECT_MANAGER_OBJECT_DIRECTORY_H
+
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "ray/id.h"
+#include "ray/raylet/mock_gcs_client.h"
+#include "ray/status.h"
+
+namespace ray {
+
+struct RemoteConnectionInfo {
+  RemoteConnectionInfo(const ClientID &id, const std::string &ipaddr, uint16_t portnum)
```

Comment:
We're using `ip_address` in most places n the codebase. Also `portnum` -> `port_num`